src/main/java/it/gov/pagopa/payhub/auth/mapper/LimitedScopeTokenMapper.java

```diff
@@ -0,0 +1,58 @@
+package it.gov.pagopa.payhub.auth.mapper;
+
+import it.gov.pagopa.payhub.auth.dto.IamUserInfoDTO;
+import it.gov.pagopa.payhub.dto.generated.BaseUserInfo;
+import it.gov.pagopa.payhub.dto.generated.LimitedScopeResource;
+import it.gov.pagopa.payhub.dto.generated.LimitedTokenRequest;
+import it.gov.pagopa.payhub.dto.generated.UserInfoLimitedScope;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LimitedScopeTokenMapper {
+
+    public UserInfoLimitedScope mapBaseUserInfoToLimitedScope(BaseUserInfo baseUserInfo, LimitedTokenRequest request) {
+        return UserInfoLimitedScope.builder()
+        		.resource(mapRequestToLimitedScope(request))
+        		.userId(baseUserInfo.getUserId())
+        		.mappedExternalUserId(baseUserInfo.getMappedExternalUserId())
+        		.fiscalCode(baseUserInfo.getFiscalCode())
+        		.familyName(baseUserInfo.getFamilyName())
+        		.name(baseUserInfo.getName())
+        		.issuer(baseUserInfo.getIssuer())
+        		.organizationAccess(baseUserInfo.getOrganizationAccess())
+        		.organizations(baseUserInfo.getOrganizations())
+        		.brokerId(baseUserInfo.getBrokerId())
+        		.brokerFiscalCode(baseUserInfo.getBrokerFiscalCode())
+        		.canManageUsers(baseUserInfo.getCanManageUsers())
+        		.systemUser(baseUserInfo.getSystemUser())
+        		.traceId(baseUserInfo.getTraceId())
+        		.type(baseUserInfo.getType())
```

Comment:
is this a metadata field used to distinguish between the 2 UserInfo models? If yes, could it be ineherited form the BaseUserInfo?